src/lib/comms/sol-lwm2m.c

```diff
@@ -4718,6 +4858,178 @@ client_bootstrap(void *data)
     return false;
 }
 
+static int
+setup_access_control_object_instance_for_instance(
+    struct sol_lwm2m_client *client,
+    uint16_t target_object_id,
+    uint16_t target_instance_id,
+    int64_t owner_server_id,
+    struct sol_lwm2m_resource *acl_res,
+    bool register_with_coap)
+{
+    struct obj_ctx *acc_obj_ctx;
+    int r;
+    struct sol_buffer buf = SOL_BUFFER_INIT_FLAGS(NULL, 0, SOL_BUFFER_FLAGS_NO_NUL_BYTE);
+    struct sol_lwm2m_resource *res;
+    struct sol_lwm2m_payload payload = { .type = SOL_LWM2M_CONTENT_TYPE_TLV };
+
+    //Only the Client itself and the owner server is able to manage
+    // Access Control Object Instances, and Security Object Instances
+    // can only be managed by Bootstrap Servers, so there's no sense in
+    // creating Access Control Object Instances or ACLs for them
+    if (target_object_id == SECURITY_SERVER_OBJECT_ID ||
+        target_object_id == ACCESS_CONTROL_OBJECT_ID) {
+        return 0;
+    }
+
+    acc_obj_ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
```

Comment:
We're sure it's not NULL, since this function (and any other access control-related stuff) is only called if `client->supports_access_control` is true (which is setup during `sol_lwm2m_client_new` if there's an object with `ACCESS_CONTROL_OBJECT_ID` in the list the client passes to that function.